Gutenberg: update npm packages to latest versions

vindl opened the pull request:
#### Changes proposed in this Pull Request

Updates Gutenberg npm packages to latest versions. 

It seems like this update will not require as much fixes as the ones before it. :)

#### Testing instructions

1. Navigate to http://calypso.localhost:3000/gutenberg/
2. Verify that there are no new errors in console.
3. Try inserting some content and verify that publishing still works.
4. Smoke test Calypso and check bundle sizes.

_On hold until new version of components package, that contains WordPress/gutenberg#10943 fix, is published._

vindl commented on client/devdocs/gutenberg-components/style.scss:
```diff
@@ -1,4 +1,4 @@
-@import '../../../node_modules/@wordpress/components/build-style/style';
+// @import '../../../node_modules/@wordpress/components/build-style/style';
```
cc @mmtr since you've worked on this. If I leave the style include here I'm running into:

```
{
  "status": 1,
  "file": ".../wp-calypso/node_modules/@wordpress/components/build-style/style.css",
  "line": 1615,
  "column": 27,
  "message": "Function shade is missing argument $percent.",
  "formatted": "Error: Function shade is missing argument $percent.\n        on line 1 of client/devdocs/gutenberg-components/style.scss\n        from line 7 of assets/stylesheets/sections/devdocs.scss\n>>       background: #0085ba shade(15%); }\n\n   --------------------------^\n"
}
```

The mentioned `shade` function has the argument `percent` passed at that line if you check the built file. Any ideas what might be causing this?

mmtr replied:
I'd say that the `shade` function shouldn't be in the generated CSS file given that it's not a CSS function AFAIK but I can see it on the original [SCSS file](https://github.com/WordPress/gutenberg/blob/61798e98fb37dfd7ee1ef5edab3c29f0fe0d8e1a/packages/components/src/date-time/style.scss#L43) (added on WordPress/gutenberg#7621). I'm sure that function should be converted to something else during the CSS build process.

I see the `shade` function is also used in other components (i.e. [`Button`](https://github.com/WordPress/gutenberg/blob/61798e98fb37dfd7ee1ef5edab3c29f0fe0d8e1a/packages/components/src/button/style.scss#L101)), but in all of them the function is an argument of the `color` function, so maybe the issue is on Gutenberg and it can be solved if we replace `background: theme(primary) shade(15%);` with `background: color(theme(primary) shade(15%));` on the original [SCSS file](https://github.com/WordPress/gutenberg/blob/61798e98fb37dfd7ee1ef5edab3c29f0fe0d8e1a/packages/components/src/date-time/style.scss#L43).

mmtr replied:
Confirmed that will solve the issue. Gutenberg uses [`postcss-color-function`](https://github.com/postcss/postcss-color-function) which provides the `color` interface and `shade` as one of their color adjusters.

I can open a PR fixing that as soon as I got access to the Gutenberg repo :)

vindl replied:
@gziolo could we add @mmtr to the repo to file the above fix? :)

mmtr replied:
I already asked for that in p1540284879000100-slack-gutenberg. I think only @mtias can add me.

gziolo replied:
Yes, him and @pento :)

vindl replied:
@mmtr if you don't want to wait feel free to start a PR from your fork too, but we'll definitely want to provide you with access for future fixes.

Also, it's curious that the same import wasn't causing issues for editor shell in Calypso:
https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/sections/gutenberg-editor.scss#L19
That's why I suspected it had something to do with differences in DevDocs setup at first.

mmtr replied:
@vindl Done! https://github.com/WordPress/gutenberg/pull/10943

vindl replied:
Thanks Miguel!

gziolo commented:
> It seems like this update will not require as much fixes as the ones before it. :)

We are close to the real API freeze, one week left :D

Copons commented:
I see two errors I haven't seen yet:

<img width="704" alt="screenshot 2018-10-23 at 11 32 13" src="https://user-images.githubusercontent.com/2070010/47354723-59dfd280-d6b7-11e8-9cd4-71e2acf2ab86.png">

First is in: `calypso/vendors~async-load-design~async-load-design-blocks~async-load-design-playground~async-load-gutenberg~81da215c.704af1af3a9082fb9a8d.min.js`

Second in: `calypso/vendors~build.83bedb58be69695ad3b0.min.js`

Other than that, though, I've smoke tested Guten, and it's working as well as before. :)

vindl commented:
Thanks for testing @Copons! The first one is already present in master, since the importing of Jetpack blocks (https://github.com/Automattic/wp-calypso/pull/27864), and it's kind of expected (translations issue).

I haven't noticed the second one before (will have to re-test), but it sounds like the problem with API path mapping. If there is some specific action that triggers it, knowing that would be helpful.

Copons commented:
> Thanks for testing @Copons! The first one is already present in master, since the importing of Jetpack blocks (#27864), and it's kind of expected (translations issue).

Oh, apparently somebody forgot to pull master, eh! 😤 

> I haven't noticed the second one before (will have to re-test), but it sounds like the problem with API path mapping. If there is some specific action that triggers it, knowing that would be helpful.

I thought it popped up upon first load, but in fact it is triggered by the sidebar.
If the window is large enough that the sidebar is open by default, the error is there on load; otherwise, it triggers when opening the sidebar.

blowery commented:
@vindl renovate will automatically generate PRs for updates to the WordPress monorepo (see #27592). In the future, could we please piggyback on those PRs instead of opening separate ones? It keeps the shrinkwrap normalized, gives us automatic release notes, and folks know where to look for module updates and makes is easier to report on the number of PRs related to managing dependencies. Feel free to continue on here and if you need any help, we're glad to assist.

Thanks much!

blowery commented:
On the CSS issue, I noticed that we're pulling in the built CSS from Gutenberg. wc-admin [uses a slightly different tact](https://github.com/woocommerce/wc-admin/blob/master/client/stylesheets/shared/_gutenberg-components.scss), pulling in the SCSS and processing it with postcss to theme out some colors. 

Is that something we'd be interested in doing for Calypso? If it is, is there an issue tracking it? Probably a bit related to #26910

vindl commented:
On hold until new version of `components` package, that contains https://github.com/WordPress/gutenberg/pull/10943 fix, is published.

vindl commented:
> Is that something we'd be interested in doing for Calypso? If it is, is there an issue tracking it? Probably a bit related to #26910

I believe we discussed this briefly at one point, and for now we are not planning to allow theming for Calypso integration (cc @gwwar @Copons please correct me if I got this wrong).

blowery commented:
@vindl it's fine here! I just wanted to make sure you were aware of the renovate PRs. :)

vindl commented:
> I thought it popped up upon first load, but in fact it is triggered by the sidebar.
If the window is large enough that the sidebar is open by default, the error is there on load; otherwise, it triggers when opening the sidebar.

@Copons the problem here seems to be that `WP_REST_Themes_Controller` was only recently merged into Gutenberg (https://github.com/WordPress/gutenberg/pull/10518), and our plugin version running on Dotcom doesn't have that update yet (failing request to `https://public-api.wordpress.com/wp/v2/sites/example.wordpress.com/themes?status=active`). I think we can ignore this error, since it will resolve itself once our backend code is updated.

vindl commented:
I updated all of the related packages again to the newest versions and introduced a couple of fixes that are needed for editor to boot (will point those out in inline comments). There is one new error regarding the rest route that probably requires a new mapping in middleware, and I'm planning to tackle that in a separate PR in order not to drag this one further. Otherwise, this is ready for another round of reviews and testing.

vindl commented on client/gutenberg/editor/controller.js:
```diff
@@ -38,7 +38,6 @@ function registerDataPlugins( userId ) {
 	const storageKey = 'WP_DATA_USER_' + userId;
 
 	use( plugins.persistence, { storageKey: storageKey } );
-	use( plugins.asyncGenerator );
```
This has been removed from the latest version of `@wordpress/data` package: https://github.com/WordPress/gutenberg/blob/master/packages/data/CHANGELOG.md#300-2018-10-29

vindl commented on client/gutenberg/editor/edit-post/components/header/index.js:
```diff
@@ -58,9 +57,6 @@ function Header( {
 						isToggled={ isEditorSidebarOpened }
 						aria-expanded={ isEditorSidebarOpened }
 					>
-						<DotTip id="core/editor.settings">
```
This was causing the editor to break, so I'm removing it since this is the only usage of `DotTip` in edit post, and we don't want to use the guided tour anyway. I'm suspecting its related to some specific error with `@wordpress/nux` package and not to our integration.
